Changes for JOSS publication (#948)
VILLASframework/node: paper/paper.md

# Summary

VILLASnode is a multi-protocol gateway, designed to facilitate real-time data exchange between various components of geographically distributed real-time experiments. Components can be testbeds, digital real-time simulators, software tools, and physical devices. It is designed for the co-simulation of power system and related energy applications.
VILLASnode was originally designed for co-simulation of electrical networks, but was further developed for a bigger variety of use cases and domains.

> Whereas distributed computing, systems, and algortihms aim to solve a common task, geographically distributed experiments link infrastructures with different components to make these components accessible to other infrastructures. Thus, data exchanges are possible that would not be possible in one single infrastructure.

Reviewer: I know this has not been changed by this PR, but I don't get this sentence. Distributed computing, systems (in general???), and algorithms solve a common task?
I'd say, we either need to rephrase this or remove it.

VILLASnode serves as the gateway that connects components across different infrastructures by providing a set of protocols and customized third-party implementations, e.g., different simulators. It enables seamless collaboration in research and testing environments while safeguarding the intellectual property of the infrastructures. The components of every infrastructure appear as a black box. The infrastructure does not need to share models or confidential information.

> VILLASnode is a Linux command line tool, which can run as installed from source or as a container. It is written in C/C++ and designed in a modular and extensible way.

Reviewer: ```suggestion
VILLASnode is a library, written in C/C++ and designed in a modular and extensible way.
```
VILLASnode is not a command line tool. `villas-node` is a command line tool, but this is only one of the binaries build by VILLASnode

![Example of modular experimental design with nodes, paths, and hooks [@villasnode_docs].](figures/VILLASnode_paths.svg)

> External software tools and programming languages can be interfaced with VILLASnode via several methods. Locally, they can be spawned as sub-processes and exchange data via shared-memory or standard I/O streams. Remotely, they can be interfaced via any of the supported node-types and protocols, such as MQTT or e.g., plain TCP sockets.

Reviewer: ```suggestion
External software tools and programming languages can be interfaced with VILLASnode via several methods. Locally, they can be spawned as sub-processes and exchange data via shared-memory or standard I/O streams. Remotely, they can be interfaced via any of the supported node-types and protocols, such as MQTT or TCP sockets.
```

Additionally, VILLASnode can be configured and controlled remotely by an HTTP REST-style Application Programming Interface (API). An OpenAPI specification of the API is provided in order to generate API bindings for a variety of different programming languages.

> Since VILLASnode is written in C/C++, it supports real-time capabilites and can supports real-time tuning for Linux systems.

Reviewer: ```suggestion
VILLASnode is written in C++ and supports real-time execution on real-time time Linux systems.
```
We use a C++ compiler, so it is C++ code. Also, the language doesn't make it real-time capable.

Helics is designed for large-scale testing and supports bindings for different languages. It uses a broker that manages the communication between the simulators, whereas VILLASnode has a peer-to-peer architecture.

# Statement of need

VILLASnode supports the coupling of real-time simulators of different hardware and software vendors, specifications, and implementations [@monti_global_2018]. These simulators play a crucial role in both academic research and industrial applications, especially within simulation of electrical power networks. They are primarily utilized for hardware-in-the-loop (HiL) simulations. As an example, in the scope of the ENSURE project, the German electrical grid is emulated with the ability to couple dynamically simulators and physical components which can be interfaced safely via analog input and output signals [@Mehlmann2023].

> This approach not only reduces development costs but also allows for the validation of components under scenarios in which it is difficult or unsafe to replicate real-world field scenarios. Moreover, VILLASnode supports the linking of simulators and real-time simulators from different vendors so that models do not need to be shared. It allows for conversation about intellectual property during collaboration. Local simulators and simulation infrastructure can be combined in a powerful arrangement unavailable at any individual infrastructure. This approach has the additional advantage of allowing co-simulation with heterogeneus methods, algorithms, and communication protocols. The communication with the components, sensors, or actuators can also be implemented using VILLASnode.

Reviewer: ```suggestion
This approach not only reduces development costs but also allows for the validation of components under scenarios in which it is difficult or unsafe to replicate real-world field scenarios. Moreover, VILLASnode supports the linking of simulators and real-time simulators from different vendors so that models do not need to be shared. It allows for communication between proprietary simulation models that cannot be shared, thus protecting intellectual property during collaboration. Local simulators and simulation infrastructure can be combined in a powerful arrangement unavailable at any individual infrastructure. This approach has the additional advantage of allowing co-simulation with heterogeneus methods, algorithms, and communication protocols. The communication with the components, sensors, or actuators can also be implemented using VILLASnode.
```

> Secondly, there is VILLASfpga, which provides hard real-time capabilities for VILLASnode.

Reviewer: ```suggestion
VILLASnode also supports interfacing with FPGAs, thus enabling hard real-time communication of software with arbitrary device connected to the FPGA.
```
VILLASfpga has been fully integrated into VILLASnode. (sorry for suggesting something that has not been touched)
